### PR TITLE
Fix header styles

### LINKED ISSUE #772: Improve grid usage
Whole template should be reviewed for incorrect usage of Bootstrap's grid system.

Some elements are misaligned or have inconsistent position in grid.

* Header (logo, search bar, cart widget) 
* cart dropdown

### PR DESCRIPTION
**Pull request by benekex2**
Fix #772

### CODE REVIEW

**Comment by artursmet** [comment]
@benekex2 Please rebase with master

**Comment by artursmet** [comment]
As we discussed on the meeting - Mobile header view needs some improvements:
* Add Menu string to Hamburger button
* Reduce padding around Salor logo, to give more space to search button

File: saleor/static/scss/storefront/components/_cart-dropdown.scss [hunk 1]
```diff
@@ -1,18 +1,65 @@
...
+          color: $dark-gray;
+        }
+      }
+      &:last-child {
+        border-bottom: 0;
+      }
+    }
+  }
+  &__total {
+    h3 {
```
Comment (weronka) [inline saleor/static/scss/storefront/components/_cart-dropdown.scss]: You can add $font-size-h3: 1rem to variables.scss. So all project will use this style for h3 elements. Also h3 is always uppercased so we can define this style for all document too.

File: saleor/static/scss/storefront/components/_cart-dropdown.scss [hunk 2]
```diff
@@ -24,21 +71,16 @@
...
-    width: 15px;
-    height: 10px;
-    position: absolute;
-    top: 47px;
-    right: 25px;
-    background-color: $light-gray;
-    clip-path: polygon(50% 0%, 0% 100%, 100% 100%);
+    .btn {
+      width: 100%;
+      padding: $global-padding/2;
```
Comment (weronka) [inline saleor/static/scss/storefront/components/_cart-dropdown.scss]: Add spaces before and after '/'. It'll be more consistent with rest scss files.

File: saleor/static/scss/storefront/components/_header.scss [hunk 3]
```diff
@@ -2,97 +2,134 @@
...
+      margin-bottom: 0;
+      color: $dark-gray;
+      line-height: 1.9rem;
+    }
+    ul {
+      list-style: none;
+      margin-bottom: 0;
+      li {
+        display: inline-block;
+        padding: $global-padding/2;
```
Comment (weronka) [inline saleor/static/scss/storefront/components/_header.scss]: spaces before and after '/'

**Pull request merged**